### DISCUSSION TITLE
Suggestions for topics for a series

### DISCUSSION THREAD

**Discussion by venkats** [java/devjava-content#24] [New Content Ideas]
### Content Details

Greetings,

I request the feedback on the topics for an article series I'd like to write. The following are a few ideas I have in mind:

Topics:
Know Your Java?
Refactoring from Imperative to Functional Style
Making use of Pattern Matching

Details:
Know Your Java?:
12-parts articles on Know your Java? The topics will include things in Java that developers may miss, the compilers may not catch, unit tests sometimes may not detect, but is useful for developers to know. In each article, the reader will look at a sample code, try to guess the behavior, will then learn about the actual behavior, and reasons, and the ways to fix or the lessons they can learn.

Refactoring from Imperative to Functional Style
12-parts article on functional thinking. In each part, the reader will start with an imperative style code, then walk thought how to approach writing that code in functional style, and see the refactoring of the code to functional style unfold.

Making use of Pattern Matching
12-parts article on pattern matching. This series will approach how readers can apply pattern matching to solve common code from traditional to more modern Java.

Please let me know your thoughts if any of these is interesting and, if so, which one you'd suggest I focus on.

Thank you.

**Comment by carimura** [comment]
All great, let's start with either Know Your Java or Refactoring, and come back to Patterns when Record Patterns and Switch get  finalize and possibly even deconstruction patterns.

**Comment by venkats** [comment]
Thank you. I will start with Refactoring from Imperative to Functional Style. Will visit the other topics later, after the series.
